cli-plugins/plugin/plugin.go

```diff
@@ -36,14 +36,8 @@ func RunPlugin(dockerCli *command.DockerCli, plugin *cobra.Command, meta manager
 	PersistentPreRunE = func(cmd *cobra.Command, _ []string) error {
 		var err error
 		persistentPreRunOnce.Do(func() {
-			cmdContext := cmd.Context()
-			// TODO: revisit and make sure this check makes sense
-			// see: https://github.com/docker/cli/pull/4599#discussion_r1422487271
-			if cmdContext == nil {
-				cmdContext = context.TODO()
-			}
-			ctx, cancel := context.WithCancel(cmdContext)
-			cmd.SetContext(ctx)
+			ctx2, cancel := context.WithCancel(cmd.Context())
```

Comment:
nit: i guess this can be called `ctx` again

Reply:
oh true